### PR TITLE
feat: add PROCEDURAL_VOTE discussion status and subject withdrawal

### PR DESCRIPTION
**Pull request by kouloumos**
## Summary

Municipal council meetings have procedural phases where the council votes on **whether to discuss or withdraw** a subject, before any substantive discussion takes place. These were being misclassified as `SUBJECT_DISCUSSION` or `OTHER`.

This PR adds:
- `PROCEDURAL_VOTE` utterance status for urgency votes (κατεπείγον) and item withdrawals (απόσυρση/αναβολή)
- `withdrawn` field on subjects — means "this subject won't be discussed in this meeting", covering both IN_AGENDA withdrawals and rejected OUT_OF_AGENDA items

### Prompt design

The prompt was developed by reviewing 47 real examples from 8 cities in the staging DB (documented in `docs/procedural-vote-examples.md`). The key insight: everything from when an out-of-agenda item is introduced until the body confirms inclusion is `PROCEDURAL_VOTE` — even if the presenter discusses the substance before the vote.

Six real-world patterns are represented: quick informal ("αν δεν υπάρχει αντίρρηση"), two-step (ένταξη then ψήφιση), formal roll call, blanket inclusion for multiple items, long presentation before vote, and rejected κατεπείγον.

## Testing

Built 6 focused test payloads from the staging DB, each covering a specific pattern (9-39 utterances, single batch, ~30s each). After verifying all patterns individually, ran a full e2e test on `zografou/jan29_2_2026` (3h meeting, multi-batch) which produced 38 PROCEDURAL_VOTE utterances with correct subject tagging and `withdrawn: true` on the withdrawn subject.

## Notes

The `docs:` commits (design document + examples) are temporary research artifacts and will be dropped before merge.

Companion PR: schemalabz/opencouncil (DB schema, UI rendering, minutes generation).

---

> [!NOTE]
> **Medium Risk**
> Touches the LLM contract (prompt + JSON schema) and core summarize data model, so mis-tagging or loss of `withdrawn` state across batches could impact downstream consumers; changes are otherwise localized with tests.
> 
> **Overview**
> Adds a new `DiscussionStatus.PROCEDURAL_VOTE` to distinguish procedural votes (urgent-item inclusion and agenda item withdrawal/postponement) from substantive discussion, and threads it through summarize output types, batch-processing JSON schema, logging, and prompt guidance.
> 
> Introduces an optional `withdrawn` flag on subjects to mark items that won’t be discussed in the meeting (withdrawn `IN_AGENDA` items or rejected `OUT_OF_AGENDA` urgencies), propagating it through compression/decompression and enrichment so it appears in final `Subject` output, with added tests covering the new status and field. Documentation is updated to reflect the new flow and doc-writing conventions.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit d7235fb376044050bc18eefbbb607b2395864835. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

<h3>Greptile Summary</h3>

Adds `PROCEDURAL_VOTE` as a new utterance discussion status and a `withdrawn` flag on subjects to represent procedural phases (κατεπείγον inclusion votes, IN_AGENDA withdrawals/postponements) that were previously misclassified. The change is additive and well-integrated end-to-end: `withdrawn` is propagated through compression, inter-batch context, enrichment, and decompression, and the prior concern about `withdrawn` dropping across batches is fixed.

- `docs/summarize.md`'s \"Utterance Status Tagging\" section still lists only the four original statuses and omits `PROCEDURAL_VOTE` (including its required `subjectId` semantics).
- The joint-discussion auto-correction at `batchProcessing.ts:280` now includes `PROCEDURAL_VOTE`, which would mis-attribute withdrawal utterances to the primary subject if a secondary subject is ever withdrawn — the `withdrawn` flag stays correct but the PROCEDURAL_VOTE utterance link is lost.

<h3>Confidence Score: 5/5</h3>

Safe to merge; both findings are P2 edge cases that don't affect the primary happy paths validated by e2e tests.

All remaining findings are P2: a doc omission and an unlikely semantic mis-attribution in a joint-discussion + withdrawal edge case. The core `withdrawn` propagation path and `PROCEDURAL_VOTE` round-trip are well-covered by new tests, the inter-batch serialization fix is present, and the e2e run on a 3-hour meeting confirmed correct behavior.

`docs/summarize.md` (missing PROCEDURAL_VOTE entry) and the joint-discussion validation block in `batchProcessing.ts`.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| src/types.ts | Adds `PROCEDURAL_VOTE` to `DiscussionStatus` enum and `withdrawn?: boolean` to `Subject`; comment for `subjectId` updated accordingly. |
| src/tasks/summarize/types.ts | Adds `withdrawn?: boolean` to `SubjectInProgress` with clear JSDoc explaining the two cases (IN_AGENDA withdrawal and OUT_OF_AGENDA rejection). |
| src/tasks/summarize/batchProcessing.ts | Integrates `PROCEDURAL_VOTE` throughout logging/display; adds `withdrawn` to inter-batch subject context (fixing previous issue); but including PROCEDURAL_VOTE in the joint-discussion auto-correction can mis-attribute withdrawal utterances for secondary subjects. |
| src/tasks/summarize/prompts.ts | Adds comprehensive Greek-language instructions for PROCEDURAL_VOTE: fundamental rule, six real-world patterns, and examples Β–Ε; also clarifies the distinction between PROCEDURAL_VOTE and VOTE. |
| src/tasks/summarize/compression.ts | Passes `withdrawn` through `compressIds` with the correct conditional spread pattern; `decompressIds` picks it up via `...s` spread. |
| src/tasks/summarize/compression.test.ts | Adds four targeted tests: PROCEDURAL_VOTE round-trip with subjectId, null subjectId, withdrawn flag preservation, and absence-means-undefined; good coverage. |
| src/tasks/summarize/utils.ts | Adds `⚖️` emoji for PROCEDURAL_VOTE and propagates `withdrawn` in `initializeSubjectsFromExisting`. |
| src/lib/subjectEnrichment.ts | Adds `withdrawn?: boolean` to `EnrichmentInput` and propagates it to the output `Subject` with the conditional spread pattern. |
| docs/summarize.md | Streamlined per new CONTRIBUTING.md doc guidelines, but the "Utterance Status Tagging" section omits the new PROCEDURAL_VOTE status. |
| src/tasks/summarize.ts | Passes `withdrawn: subject.withdrawn` into enrichment input; correct since `EnrichmentInput.withdrawn` accepts `undefined`. |

</details>

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[LLM returns subjects + utteranceStatuses] --> B{subject has withdrawn?}
    B -- yes --> C[SubjectInProgress.withdrawn = true]
    B -- no --> D[withdrawn absent]

    C --> E[Inter-batch: serialized with withdrawn: true]
    D --> F[Inter-batch: withdrawn omitted]

    E & F --> G[Next batch LLM sees current subject list]

    A --> H{utteranceStatus.status?}
    H -- PROCEDURAL_VOTE --> I[subjectId REQUIRED]
    H -- SUBJECT_DISCUSSION/VOTE --> J[subjectId REQUIRED]
    H -- ATTENDANCE/OTHER --> K[subjectId null]

    I --> L{subject.discussedIn set?}
    L -- yes secondary --> M[⚠️ Auto-correct → primary subjectId]
    L -- no primary --> N[keep original subjectId]

    G --> O[Phase 1.5 mergeSubjects]
    O --> P[Phase 2: speakerContributions]
    P --> Q{subject.withdrawn?}
    Q -- true --> R[0 SUBJECT_DISCUSSION utterances → skip LLM call]
    Q -- false --> S[generate contributions normally]

    R & S --> T[Phase 3: enrichSubjectData]
    T --> U[withdrawn propagated to Subject output]
```

<details><summary><h3>Comments Outside Diff (1)</h3></summary>

1. `src/tasks/summarize/batchProcessing.ts`, line 484-493 ([link](https://github.com/schemalabz/opencouncil-tasks/blob/6bbd2983b70d36b071764632d23411df0544b6fc/src/tasks/summarize/batchProcessing.ts#L484-L493)) 

   <a href="#">P1</a> **`withdrawn` flag dropped from inter-batch subject context**

   The subject list serialized into the LLM prompt for each subsequent batch omits `withdrawn`, so any subject marked `withdrawn: true` in batch N is invisible as "withdrawn" to the LLM in batch N+1. If the LLM decides to include that subject in its response (e.g., it re-annotates or references an earlier subject), it will return the subject without `withdrawn`, and the accumulated `conversationState.subjects` will silently lose the flag. The "undiscussed subjects" preservation path does spread the full object, so the flag survives only when the LLM ignores the subject entirely — a fragile guarantee for 3h multi-batch meetings.

   

   <details><summary>Prompt To Fix With AI</summary>

   `````markdown
   This is a comment left during a code review.
   Path: src/tasks/summarize/batchProcessing.ts
   Line: 484-493

   Comment:
   **`withdrawn` flag dropped from inter-batch subject context**

   The subject list serialized into the LLM prompt for each subsequent batch omits `withdrawn`, so any subject marked `withdrawn: true` in batch N is invisible as "withdrawn" to the LLM in batch N+1. If the LLM decides to include that subject in its response (e.g., it re-annotates or references an earlier subject), it will return the subject without `withdrawn`, and the accumulated `conversationState.subjects` will silently lose the flag. The "undiscussed subjects" preservation path does spread the full object, so the flag survives only when the LLM ignores the subject entirely — a fragile guarantee for 3h multi-batch meetings.

   

   How can I resolve this? If you propose a fix, please make it concise.
   `````
   </details>
</details>

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
This is a comment left during a code review.
Path: docs/summarize.md
Line: 185-190

Comment:
**`PROCEDURAL_VOTE` missing from Utterance Status Tagging section**

The doc was updated but the new status doesn't appear in the "Utterance Status Tagging" key concepts section. A contributor reading this doc will still see only four statuses and won't know `PROCEDURAL_VOTE` exists or that it requires a non-null `subjectId`. Consider adding a bullet like the others.

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: src/tasks/summarize/batchProcessing.ts
Line: 278-305

Comment:
**`PROCEDURAL_VOTE` auto-correction breaks withdrawal attribution for secondary subjects**

Adding `PROCEDURAL_VOTE` to the joint-discussion auto-correction logic means that if a secondary subject (one with `discussedIn` set) is being withdrawn, the PROCEDURAL_VOTE utterances referencing it will be silently repointed to the primary subject. The `withdrawn: true` flag stays on the secondary, but the PROCEDURAL_VOTE utterances end up attributed to the primary — severing the link between the withdrawal utterances and the subject actually being withdrawn.

This is unlikely in practice (withdrawing a subject that's already in a joint-discussion group), but `SUBJECT_DISCUSSION`/`VOTE` corrections make sense because joint discussion semantics consolidate _substantive_ content to the primary. `PROCEDURAL_VOTE` for a withdrawal is the exact opposite — it is meaningful only in the context of the specific subject being withdrawn.

How can I resolve this? If you propose a fix, please make it concise.
`````

</details>

<sub>Reviews (3): Last reviewed commit: ["feat: add PROCEDURAL\_VOTE discussion sta..."](https://github.com/schemalabz/opencouncil-tasks/commit/d7235fb376044050bc18eefbbb607b2395864835) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=28874898)</sub>

### CODE REVIEW

**Comment by kouloumos** [comment]
I [force-pushed](https://github.com/schemalabz/opencouncil-tasks/compare/6bbd2983b70d36b071764632d23411df0544b6fc..241a0de0bf0210d236289143788c1bd9025a19b8) to address bot review comment(s) ([1](https://github.com/schemalabz/opencouncil-tasks/pull/37#discussion_r3105794619), [2](https://github.com/schemalabz/opencouncil-tasks/pull/37#discussion_r3105794620)):

- Include `withdrawn` in inter-batch subject serialization and prompt instruction
- Fix contradictory OUT_OF_AGENDA description to match the "always create" rule at line 144

File: src/tasks/summarize/batchProcessing.ts [hunk 1]
```diff
@@ -517,6 +532,7 @@ ${JSON.stringify(conversationState.subjects.map(s => ({
                                 locationText: { type: ["string", "null"] },
                                 topicLabel: { type: ["string", "null"] },
                                 discussedIn: { type: ["string", "null"] },
+                                withdrawn: { type: "boolean" },
```
[the remark this answers is not in the document]
Comment (kouloumos) [reply src/tasks/summarize/batchProcessing.ts:536]: Fixed in [force-push](https://github.com/schemalabz/opencouncil-tasks/compare/6bbd2983b70d36b071764632d23411df0544b6fc..241a0de0bf0210d236289143788c1bd9025a19b8) — added `withdrawn` to the serialized subject map (conditionally, only when true) and updated the prompt instruction to include it in the preservation list.

File: src/tasks/summarize/prompts.ts [hunk 2]
```diff
@@ -179,7 +190,12 @@ ${formattedTopicLabels}
 type:
 - IN_AGENDA: Θέμα από ημερήσια διάταξη (ΗΔΗ ΥΠΑΡΧΕΙ - μην δημιουργείς νέο!)
 - BEFORE_AGENDA: Προ-ημερησίας διάταξη συζήτηση (ΔΗΜΙΟΥΡΓΗΣΕ νέο subject για ανακοινώσεις, ερωτήματα)
-- OUT_OF_AGENDA: Εκτός διάταξης θέματα προς ψήφιση (ΔΗΜΙΟΥΡΓΗΣΕ νέο subject για επείγοντα που δεν ήταν στην αρχική διάταξη)
+- OUT_OF_AGENDA: Εκτός διάταξης θέματα προς ψήφιση (ΔΗΜΙΟΥΡΓΗΣΕ νέο subject για επείγοντα που ΕΓΚΡΙΘΗΚΑΝ ως κατεπείγοντα)
```
[the remark this answers is not in the document]
Comment (kouloumos) [reply src/tasks/summarize/prompts.ts]: Fixed in [force-push](https://github.com/schemalabz/opencouncil-tasks/compare/6bbd2983b70d36b071764632d23411df0544b6fc..241a0de0bf0210d236289143788c1bd9025a19b8) — updated the OUT_OF_AGENDA type description to say "για κάθε κατεπείγον, είτε εγκριθεί είτε απορριφθεί", consistent with the detailed instructions above.

File: src/tasks/summarize/batchProcessing.ts [hunk 3]
```diff
@@ -517,6 +533,7 @@ ${JSON.stringify(conversationState.subjects.map(s => ({
...
```
[the remark this answers is not in the document]
Comment (kouloumos) [reply src/tasks/summarize/batchProcessing.ts:536]: Out of scope — the root cause (withdrawn missing from inter-batch serialization) was already fixed in the prior force-push. Now that the LLM sees `withdrawn: true` on subjects, and withdrawn subjects don't generate new discussion, there's no realistic scenario where the LLM would actively return `withdrawn: false`. The undiscussed-subjects guard already preserves subjects the LLM omits entirely.

**Comment by kouloumos** [comment]
Squashed the 3 commits into a single clean commit. Also added documentation guidelines to CONTRIBUTING.md and simplified `docs/summarize.md`.

Dropped commits (still accessible by hash):
- 5b8ece9 — `docs: add procedural votes design document`
- 241a0de — `docs: add procedural vote examples from staging DB review`

Ready to merge.
